azure-kusto-data/azure/kusto/data/_telemetry.py

```diff
@@ -0,0 +1,158 @@
+from typing import Callable, Optional
+
+from azure.core.settings import settings
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.tracing.decorator_async import distributed_trace_async
+from azure.core.tracing import SpanKind
+
+from .client_request_properties import ClientRequestProperties
+
+
+class KustoTracingAttributes:
+    """
+    Additional ADX attributes for telemetry spans
+    """
+
+    _KUSTO_CLUSTER = "Kusto Cluster"
+    _DATABASE = "Database"
+    _TABLE = "Table"
+
+    _AUTH_METHOD = "Authentication Method"
+    _CLIENT_ACTIVITY_ID = "Client Activity ID"
+
+    _SPAN_COMPONENT = "component"
+    _HTTP = "http"
+    _HTTP_USER_AGENT = "http.user_agent"
+    _HTTP_METHOD = "http.method"
+    _HTTP_URL = "http.url"
+
+    @classmethod
+    def add_attributes(cls, **kwargs) -> None:
+        """
+        Add ADX attributes to the current span
+        :key dict tracing_attributes: key, val ADX attributes to include in span of trace
+        """
+        tracing_attributes: dict = kwargs.pop("tracing_attributes", {})
```

Comment:
if span_impl_type is None:
    return

To avoid nesting

Reply:
see next line:
if none - skips next line and returns

Reply:
I meant change:
```python
        if span_impl_type is not None:
            current_span = span_impl_type.get_current_span()
            span = span_impl_type(span=current_span)
            for key, val in tracing_attributes.items():
                span.add_attribute(key, val)
```
to:
```python
        if span_impl_type is None:
            return

        current_span = span_impl_type.get_current_span()
        span = span_impl_type(span=current_span)
        for key, val in tracing_attributes.items():
            span.add_attribute(key, val)
```